.devcontainer/README.md

```diff
@@ -0,0 +1,217 @@
+
+### **6. Working Inside the Devcontainer**
+
+With the container launched, you're now in an environment tailored for CCCL development. Whether coding, running tests, or conducting other tasks, you'll benefit from the tailored tools and settings of this environment.
+
+For more information on building and running tests, see the [contributer guide](../CONTRIBUTING.md).
+
+## Using Docker Images Directly (Manual Approach)
+
+While Visual Studio Code with Dev Containers offers a smooth development experience, there might be developers who lean towards a direct Docker approach. For those, we've simplified the process with our `launch.sh` script.
+
+Before using the script, you might want to explore the available configurations.
+The `.devcontainer` directory and its subdirectories house these configurations, with each subdirectory typically named as `cuda<CTK_VERSION>-<HOST-COMPILER>`.
+
+### **Steps**:
+
+1. **Use the `launch.sh` Script**:
+   To initiate your desired development environment, run the `launch.sh` script with flags specifying your desired CUDA and compiler versions:
+   ```bash
+   ./launch.sh --cuda 12.2 --host gcc12 --docker
+   ```
+   This command pulls the appropriate Docker image, runs it, and provides an interactive shell within the container.
+
+2. **Workspace Setup**:
+   The `launch.sh` script automatically mounts the CCCL repository directory into the container. This ensures you have a mirrored workspace inside the container, resembling your local setup.
+
+3. **Working Inside the Container**:
+   Inside the container, you are equipped with all tools and configurations designed for CCCL development. You can initiate builds, run tests, or perform any other development tasks as you would in a local setup.
+
+4. **Exiting the Container**:
+   After you're finished with your tasks, simply type `exit` in the interactive shell to leave. Upon exit, the container gets removed, assuring a pristine state for your next session.
+
+## `launch.sh`: Development Container Launcher Script
+
+### Overview
+
+This script provides a streamlined process to launch a development container environment tailored for CCCL development.
+The environment can be started either in Docker directly or within Visual Studio Code (VSCode) using its "Remote - Containers" feature.
+
+### Usage
+
+```bash
+./launch.sh [-c|--cuda <CUDA version>] [-H|--host <Host compiler>] [-d|--docker]
+```
+
+### Options
+
+- `-c, --cuda`: Specify the CUDA version for the environment. For example, `12.2`.
+- `-H, --host`: Specify the host compiler version for the environment. For example, `gcc12`.
+- `-d, --docker`: By default, the script assumes you want to launch the environment in VSCode. Use this option if you want to run the development environment in Docker directly.
```

Comment:
I wonder if this flag should be flipped. IIUC VSCode users will mostly be coming from the VSCode GUI, so the default `launch.sh` user is probably going to be going for raw Docker right?

Reply:
I'd been considering the exact same thing. The fact that you had the same idea tells me I should probably make that change :slightly_smiling_face:.